File: app/src/main/java/by/vshkl/bashq/common/Navigator.java
package by.vshkl.bashq.common;

import android.content.Context;
import android.content.Intent;
import android.text.Html;

import javax.inject.Inject;

import by.vshkl.bashq.injection.scope.PerActivity;
import by.vshkl.bashq.ui.acticity.ComicsActivity;

@PerActivity
public class Navigator {

    @Inject
    public Navigator() {
    }

    public void navigateToComicsActivity(Context context) {
        if (context != null) {
            context.startActivity(ComicsActivity.getCallingIntent(context));
        }
    }

    public void navigateToQuoteShareLinkChooser(Context context, String quoteLink) {
        if (context != null) {
            Intent shareLinkIntent = new Intent();
            shareLinkIntent.setAction(Intent.ACTION_SEND);
            shareLinkIntent.setType("text/plain");
            shareLinkIntent.putExtra(Intent.EXTRA_TEXT, quoteLink);
            context.startActivity(Intent.createChooser(shareLinkIntent, "Share quote link with..."));
        }
    }

    public void navigateToQuoteShareTextChooser(Context context, String quoteText) {
        if (context != null) {
            Intent shareTextIntent = new Intent();
            shareTextIntent.setAction(Intent.ACTION_SEND);
            shareTextIntent.setType("text/plain");
            shareTextIntent.putExtra(Intent.EXTRA_TEXT, Html.fromHtml(quoteText).toString());
            context.startActivity(Intent.createChooser(shareTextIntent, "Share quote text with..."));
        }
    }

    public void navigateToComicsShareImageChooser(Context context, String comicsImageLink) {
        if (context != null) {
            Intent shareTextIntent = new Intent();
            shareTextIntent.setAction(Intent.ACTION_SEND);
            shareTextIntent.setType("text/plain");
            shareTextIntent.putExtra(Intent.EXTRA_TEXT, comicsImageLink);
            context.startActivity(Intent.createChooser(shareTextIntent, "Share comics with..."));
        }
    }
}

File: app/src/main/java/by/vshkl/bashq/common/Navigator.kt
package by.vshkl.bashq.common

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.support.v4.app.ActivityOptionsCompat
import android.support.v7.app.AppCompatActivity
import android.view.View
import by.vshkl.bashq.R
import by.vshkl.bashq.ui.activity.ComicActivity
import by.vshkl.bashq.ui.activity.GalleryActivity

class Navigator {

    companion object {
        fun navigateToGalleryActivity(context: Context) {
            val intent = GalleryActivity.getCallingIntent(context)

            context.startActivity(intent)
        }

        fun navigateToComicViewActivity(context: Context, comicUrl: String, view: View) {
            val intent = ComicActivity.getCallingIntent(context)
            intent.putExtra(GalleryActivity.EXTRA_COMIC_URL, comicUrl)

            val sharedView = view
            val transitionName = context.getString(R.string.comic_view_transition)
            val transitionActivityOptions = ActivityOptionsCompat.makeSceneTransitionAnimation(
                    context as AppCompatActivity, sharedView, transitionName)

            context.startActivity(intent, transitionActivityOptions.toBundle())
        }

        fun navigateToShareQuoteChooser(context: Context, quoteText: String) {
            val intent = Intent()
            intent.action = Intent.ACTION_SEND
            intent.putExtra(Intent.EXTRA_TEXT, quoteText)
            intent.type = "text/plain"

            context.startActivity(Intent.createChooser(intent, "Share quote..."))
        }

        fun navigateToShareComicChooser(context: Context, quoteText: String, imageUri: Uri) {
            val intent = Intent()
            intent.action = Intent.ACTION_SEND
            intent.putExtra(Intent.EXTRA_TEXT, quoteText)
            intent.putExtra(Intent.EXTRA_STREAM, imageUri)
            intent.type = "image/*"
            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)

            context.startActivity(Intent.createChooser(intent, "Share comic..."))
        }
    }
}
